src/main/java/hu/akarnokd/rxjava2/operators/FlowableWindowPredicate.java

```diff
@@ -92,6 +92,8 @@ protected void subscribeActual(Subscriber<? super Flowable<T>> s) {
 
         UnicastProcessor<T> window;
 
+        T pending; // look-ahead element for Mode.BEFORE
```

Comment:
There was a similar problem with `scan` that caused unwanted element delay. You may need to have full queue-drain on the window-emission side to reliably solve the problem, similar to `groupBy`, because we can't know if the next element will trigger a new window or not.

Reply:
Ah good tip to look at GroupBy.
We have a much simpler problem here though, as there is only ever one active window.
When finding the element that toggles the predicate, no element is requested from upstream, so at that point the number of upstream requests is equal to the number of requested windows downstream. Because of this we can safely emit a new window in the onNext call.
The only tricky thing is that the BEFORE split mode (windowUntil) will emit 2 elements to the newly created window at once.
We could try to check the requested number of downstream windows and emit the window immediately if allowed, but we'd still have to defer the emission of the element when no more windows were requested.

Reply:
If there is an element held in `pending` due to the case you described, a request from downstream should trigger the emission of a window immediately since the upstream before the operator may not emit the next item in a long time or ever. For this you need the full queue-drain trampolining found in many operators.